File: frontend/src/api/apply.ts
import { apiPost } from '@/lib/api-client';

export interface ApplicantInfo {
  first_name: string;
  last_name: string;
  email: string;
  phone?: string;
}

export interface PrepareResponse {
  ats_type: string | null;
  ats_detected: boolean;
  cover_letter: string | null;
  resume_tweaks: Record<string, unknown> | null;
  applicant_info: ApplicantInfo;
  job_title: string;
  company: string;
  job_url: string;
}

export interface KitRequest {
  /** user-edited cover letter, saved before the kit builds */
  cover_letter?: string;
}

/** The application kit: everything prefilled, YOU send it.
    Questboard never transmits an application. */
export interface KitResponse {
  success: boolean;
  method: string | null;
  message: string;
  /** where the human acts */
  apply_url: string;
  /** the ATS form, prefilled (empty for unmapped ATS types) */
  fields: Record<string, string>;
}

export function prepareApplication(id: number): Promise<PrepareResponse> {
  return apiPost<PrepareResponse>(`/applications/${id}/prepare`);
}

export function getApplicationKit(id: number, data: KitRequest = {}): Promise<KitResponse> {
  return apiPost<KitResponse>(`/applications/${id}/apply`, data);
}

File: backend/app/schemas/apply.py
from __future__ import annotations

from pydantic import BaseModel


class PrepareResponse(BaseModel):
    ats_type: str | None = None  # "greenhouse", "lever", None
    ats_detected: bool
    cover_letter: str | None = None
    resume_tweaks: dict | None = None
    applicant_info: dict  # { first_name, last_name, email, phone }
    job_title: str
    company: str
    job_url: str


class KitRequest(BaseModel):
    cover_letter: str | None = None  # User-edited version, saved before the kit builds


class KitResponse(BaseModel):
    """The application kit: everything prefilled, the human sends it.

    Questboard never transmits an application (docs/anti-slop.md).
    """
    success: bool
    method: str | None = None  # "greenhouse", "lever", "linkedin", None
    message: str
    # where the human acts
    apply_url: str = ""
    # the ATS form, prefilled (empty for unmapped ATS types)
    fields: dict = {}
